alf/config_util.py

```diff
@@ -117,6 +119,13 @@ def func(self, a, b):
             This flag overrides the mutable flag if True. For users wanting this
             to be the default behavior, the ALF_SOLE_CONFIG env variable can be
             set to 1.
+        override_all (bool): If True, the value of the config will be set regardless
+            of any preeixsting mutable or sole_init settings. This should be used
+            only when absolutely necessary (e.g., a teacher-student training loop,
+            where the student must override certain configs inheritted from the
+            teacher). Otherwise, use ``mutable`` or ``sole_init`` instead.
+            NOTE: if override_all is True, only the value of the config will be
```

Comment:
I don't quite understand this comment. Why "only"? What is the possible alternative?
And why is "_all" suffix needed? What does "all" mean here?

Reply:
The other alternative is that `alf.config(..., override_all=True)` will also set the `mutable` and `sole_init` flags of that particular config along with the value. This is probably not what the user wants though. As an example, imagine that a user sets a config value to be immutable. The user then wants to one-time override that value by calling `alf.config(..., override_all=True)`. This call would set `mutable=True` by default, which has now changed that config's original setting. To me, `alf.config(..., override_all=True)` should only be for changing the value of a config and respect other parameters since it's used in very nuanced situations. The other flipside is that we allow the call to modify `mutable` and `sole_init`, but if the user wanted to respect the previous settings, it would require them to somehow find them out.

The `_all` suffix is meant to mean that we change the value of a config regardless of "all" of its protections: `mutable` and `sole_init`.

Reply:
> The other alternative is that `alf.config(..., override_all=True)` will also set the `mutable` and `sole_init` flags of that particular config along with the value. This is probably not what the user wants though. As an example, imagine that a user sets a config value to be immutable. The user then wants to one-time override that value by calling `alf.config(..., override_all=True)`. This call would set `mutable=True` by default, which has now changed that config's original setting. To me, `alf.config(..., override_all=True)` should only be for changing the value of a config and respect other parameters since it's used in very nuanced situations. The other flipside is that we allow the call to modify `mutable` and `sole_init`, but if the user wanted to respect the previous settings, it would require them to somehow find them out.

How about just explicitly comment that overrid_all=True does not change the value of mutable or sole_init? This is clearer to me.

> 
> The `_all` suffix is meant to mean that we change the value of a config regardless of "all" of its protections: `mutable` and `sole_init`.

`_all` suffix seems unnecessary.  "override" would mean the same thing.

Reply:
Sounds good. Done.